rest/config.go

```diff
@@ -837,3 +852,33 @@ func ServerMain(runMode SyncGatewayRunMode) {
 	ValidateConfigOrPanic(runMode)
 	RunServer(config)
 }
+
+// localhostAddr takes an address like ':4984' or '127.0.0.1:4984' and prefixes with localhost if required.
+func localhostAddr(addr string) string {
+	port := regexp.MustCompile("^:\\d+$")
+	if port.MatchString(addr) {
+		return "localhost" + addr
+	}
+	return addr
+}
+
+// waitForResponse blocks until the given address returns a HTTP 200 response at the root
+func waitForResponse(addr string) error {
+	// TODO: Could this ever be https-only?
+	url := "http://" + localhostAddr(addr) + "/"
+
+	err, _ := base.RetryLoop(
+		"GET "+url,
+
+		base.RetryWorker(func() (bool, error, interface{}) {
+			resp, err := http.Get(url)
+			shouldRetry := err != nil || resp.StatusCode != http.StatusOK
+			return shouldRetry, nil, nil
+		}),
+
+		// 100ms * 64^2 == ~6.8 minutes longest wait
```

Comment:
I believe this is 2^64, not 64^2.  So you'll want to dial this back to something like 12.